### PR TITLE
Trailz function added to the fortran compiler

### PR DESCRIPTION
**Pull request by CarolineConcatto**
Author: Caroline Concatto <caroline.concatto@arm.com>
TRAILZ returns the number of trailing zero bits of an integer.

### CODE REVIEW

**Comment by kiranchandramohan** [comment]
@CarolineConcatto can you rebase?

File: runtime/flang/ktrailz.c [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+int64_t
+__mth_i_ktrailz(int64_t i)
+{
+  uint64_t ui=i; /* unsigned representation of 'i' */
+
+ #if (defined(PGOCL) || defined(TARGET_LLVM_ARM)) && !defined(TARGET_LLVM_ARM64)
+  return (ui)? __builtin_ctz(ui):64;
+ #else
+   if (!ui) return 64;
+   return ((int)(ui))?( __builtin_ctz(ui)): (__builtin_ctz(ui>>32+32));
```
Comment (shivaramaarao) [inline runtime/flang/ktrailz.c:28]: +32 need to be outside the bracket here
(__builtin_ctz(ui>>32))+32

**Review by shivaramaarao — APPROVED** [review]

File: tools/flang2/utils/ilmtp/ppc64le/ilmtp.n [hunk 2]
```diff
@@ -2820,6 +2820,19 @@ Cast an unsigned integer to a 64-bit integer.
 64-bit integer LEADZ intrinsic
 .AT i8
 .OP KLEADZ r p1
+.IL TRAILZ intr lnk
```
[the remark this answers is not in the document]
Comment (gklimowicz) [reply tools/flang2/utils/ilmtp/ppc64le/ilmtp.n:2823]: Still doesn't build on Power without this change...

**Review by gklimowicz — CHANGES_REQUESTED** [review]

File: tools/flang2/utils/ilmtp/ppc64le/ilmtp.n (hunk 2 above)
[the remark this answers is not in the document]
Comment (kiranchandramohan) [reply tools/flang2/utils/ilmtp/ppc64le/ilmtp.n:2823]: You mean it still **doesn't** build with this change?

[the remark this answers is not in the document]
Comment (gklimowicz) [reply tools/flang2/utils/ilmtp/ppc64le/ilmtp.n:2823]: No, still doesn't compile.

I think it needs to be **B**TRAILZ, like it is in `tools/flang2/utils/ilmtp/aarch64/ilmtp.n` and `tools/flang2/utils/ilmtp/x86_64/ilmtp.n`

[the remark this answers is not in the document]
Comment (CarolineConcatto) [reply tools/flang2/utils/ilmtp/ppc64le/ilmtp.n]: Thank you @gklimowicz. 
Can you build again with the change to see if it works?

**Review by gklimowicz — APPROVED** [review]
Builds on OpenPOWER now. Thanks, Carol.

**Pull request merged**